### ISSUE TITLE
URL is not being shortened correctly..

### ISSUE THREAD

**Issue by shrey2199** [YOURLS/YOURLS#2986]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/YOURLS/.github/blob/main/CODE_OF_CONDUCT.md).

### Submission validity

- [X] This is not a personal support request, that should be posted on [discussions community](https://github.com/YOURLS/YOURLS/discussions).
- [X] I checked [current issues](https://github.com/YOURLS/YOURLS/issues) and this request isn't a duplicate of an existing issue, opened or closed.

### Self troubleshooting

- [X] I've read the [documentation](https://yourls.org/) and made sure to follow it.
- [X] I've read the [troubleshooting first steps](https://github.com/YOURLS/YOURLS/wiki/Troubleshooting-first-steps) guide.

### Version

1.8.1

### Description

I was trying to shorten some URLs but they were not shortened correctly

eg. 

For this URL https://index.shrey23.workers.dev/1:/MIRROR%20BOT/South%20Dubbed%20Movies/Rangasthalam.mkv

the shortened url redirected to https://index.shrey23.workers.dev/1:/MIRRORBOT/SouthDubbedMovies/Rangasthalam.mkv

also 

in this URL : https://lib.shrey23.eu.org/api/v1/redirectdownload/Rangasthalam.mkv?a=zv2b0tminkluwdbbk2zs52ewozq7g28tfvqd615lonhk9ff7yc&id=1xYtdWa3C-zez3VLG_4j0MiHTsAOM3FXe

the short url redirects to https://lib.shrey23.eu.org/api/v1/redirectdownload/Rangasthalam.mkv?a=zv2b0tminkluwdbbk2zs52ewozq7g28tfvqd615lonhk9ff7yc which causes not found error. 

I have checked previous issues but the problem was not that the spaces were getting completely removed like in 1st case and the url after & symbol not being accepted.

All these URLs work when added in Admin Panel but they don't work when added through API
 PLs Help !!

**Comment by matthewpkemp** [comment]
I just attempted the same shortening w/ URL using Short Menu on my domain (1.8.1) and the shortening occurs correctly.  Curious as to how you are calling using the API (what tool/script you are using) as it may be sanitizing the input before the API call.

**Comment by shrey2199** [comment]
@matthewpkemp Short Menu as in admin panel ?? There it works for me as well but when i try to do it with api the errors occur

i use : https://domain/yourls-api.php?signature=SIG&action=shorturl&url=LINK&keyword=WORD&format=json

**Comment by matthewpkemp** [comment]
@shrey2199 Short Menu is an app <https://shortmenu.com>.  Are you hitting the url from a browser, or code that you have written?

**Comment by shrey2199** [comment]
@matthewpkemp Browser and Python code both and i am facing this error in both

Some applications have native encoding and decoding support maybe that is the reason it works on shortmenu

**Comment by matthewpkemp** [comment]
@shrey2199 I believe so:

Short Menu:  "GET /yourls-api.php?format=json&signature=XXXXXX&url=https%3A%2F%2Findex.shrey23.workers.dev%2F1%3A%2FMIRROR%2520BOT%2FSouth%2520Dubbed%2520Movies%2FRangasthalam.mkv&action=shorturl HTTP/2.0" 200 887 "-" "Short%20Menu/603 CFNetwork/1128.1 Darwin/19.6.0 (x86_64)"

Safari: "GET /yourls-api.php?signature=XXXXXX&action=shorturl&url=https://index.shrey23.workers.dev/1:/MIRROR%20BOT/South%20Dubbed%20Movies/Rangasthalam.mkv HTTP/2.0" 200 804 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1.1 Safari/605.1.15"

Basically Short Menu is handling the encoding ex: %20 into %2520 in order to pass it correctly to the API.  I would say that the API is handling this as expected by taking what is passed to it and storing it.

**Comment by shrey2199** [comment]
@matthewpkemp Thanks For Telling me that, I implemented URL encoding through urllib in my python script and now it works very well.

**Comment by matthewpkemp** [comment]
@shrey2199 good news.  Perhaps we can mark this as closed?  We may need one of the maintainers to do so, not sure if you can as you originally reported the bug.
